# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/StructuredCloneHolder.h
# Commit: 9d5a2c2147a1
# Full Hash: 9d5a2c2147a1e2b22c94dd069f1d5dcb92708c4d
# Author: Paul Adenot <paul@paul.cx>
# Date: 2024-03-07 09:55:57
# Regressor Bug: 1858958
# File Overlap Count: 1
# Description:
#   Bug 1858958 - AudioData: structured cloning. r=chunmin,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D190948
# ==============================================================================

diff -r fdc3b5b9ac9e -r 9d5a2c2147a1 dom/base/StructuredCloneHolder.h
--- a/dom/base/StructuredCloneHolder.h	Wed Mar 06 14:00:14 2024 +0000
+++ b/dom/base/StructuredCloneHolder.h	Wed Mar 06 14:00:14 2024 +0000
@@ -169,6 +169,7 @@
 class MessagePort;
 class MessagePortIdentifier;
 struct VideoFrameSerializedData;
+struct AudioDataSerializedData;
 
 class StructuredCloneHolder : public StructuredCloneHolderBase {
  public:
@@ -270,6 +271,8 @@
 
   nsTArray<VideoFrameSerializedData>& VideoFrames() { return mVideoFrames; }
 
+  nsTArray<AudioDataSerializedData>& AudioData() { return mAudioData; }
+
   nsTArray<EncodedVideoChunkData>& EncodedVideoChunks() {
     return mEncodedVideoChunks;
   }
@@ -379,6 +382,9 @@
   // Used for cloning VideoFrame in the structured cloning algorithm.
   nsTArray<VideoFrameSerializedData> mVideoFrames;
 
+  // Used for cloning AudioData in the structured cloning algorithm.
+  nsTArray<AudioDataSerializedData> mAudioData;
+
   // Used for cloning EncodedVideoChunk in the structured cloning algorithm.
   nsTArray<EncodedVideoChunkData> mEncodedVideoChunks;
 